amundsen_application/static/js/components/common/SearchBar/InlineSearchResults/ResultItemList/index.tsx

```diff
@@ -56,10 +58,13 @@ class ResultItemList extends React.Component<ResultItemListProps, {}> {
   }
 
   render = () => {
-    const { resourceType, suggestedResults, title } = this.props;
+    const { resourceType, suggestedResults, title, showBetaFlag } = this.props;
     return (
       <>
-        <div className="section-title title-3">{title}</div>
+        <div className="section-title title-3">
+          { title }
+          { showBetaFlag && <Flag text="beta" labelStyle="default"/> }
```

Comment:
We could opt to check the `resourceType === ResourceType.dashboard` to show the `Flag`.